src/controllers/controller.bar.js

```diff
@@ -478,11 +484,13 @@ module.exports = function(Chart) {
 			var index = rectangle._index;
 			var custom = rectangle.custom || {};
 			var model = rectangle._model;
-			var rectangleElementOptions = this.chart.options.elements.rectangle;
+			var options = rectangle.$previousStyle ? rectangle.$previousStyle : this.chart.options.elements.rectangle;
```

Comment:
do you need to check if `$previousStyle` is defined here? it would seem to me that there's no way for it to be undefined